File: client/cs/src/Proton.Drive.Sdk/Telemetry/UploadSizeClass.cs
namespace Proton.Drive.Sdk.Telemetry;

// Numeric values must match the UploadSizeClass enum in proton.drive.sdk.proto.
public enum UploadSizeClass
{
    Unspecified = 0,
    Small = 1,
    Single = 2,
    Multi = 3,
}

File: incubating/client/kt/sdk/src/main/kotlin/me/proton/drive/sdk/extension/UploadSizeClass.kt
package me.proton.drive.sdk.extension

import me.proton.drive.sdk.telemetry.UploadSizeClass
import proton.drive.sdk.ProtonDriveSdk

fun ProtonDriveSdk.UploadSizeClass.toEnum() = when(this) {
    ProtonDriveSdk.UploadSizeClass.UPLOAD_SIZE_CLASS_UNSPECIFIED -> UploadSizeClass.UNSPECIFIED
    ProtonDriveSdk.UploadSizeClass.UPLOAD_SIZE_CLASS_SMALL -> UploadSizeClass.SMALL
    ProtonDriveSdk.UploadSizeClass.UPLOAD_SIZE_CLASS_SINGLE -> UploadSizeClass.SINGLE
    ProtonDriveSdk.UploadSizeClass.UPLOAD_SIZE_CLASS_MULTI -> UploadSizeClass.MULTI
    ProtonDriveSdk.UploadSizeClass.UNRECOGNIZED -> UploadSizeClass.UNRECOGNIZED
}
